readme.md

```diff
@@ -532,53 +617,37 @@ interface PaymentConfirmation {
 }
 ```
 
+#### 4.b Optional: Extension rejects payment
 
-#### 5. Attester submits credential
+The extension MUST send a [rejection message](#rejections) if the user cancels or rejects the payment.
 
-|||
-|-|-|
-| direction | `dApp -> extension` |
-| message_type | `'submit-attestation'` |
-
-```typescript
-interface Attestation {
-    /** same as the `rootHash` value of the `'request-attestation'` message */
-    claimHash: string
 
-    /** Hash of the CType*/
-    cTypeHash: string
+#### 5.a Attester submits credential
```

Comment:
I think we should also allow sending only this message to allow scenarios where everything from entering and confirming claim data to selecting a DID to be used (ideally also requesting a challenge to be signed with that DID) is handled in an app outside the the credential api flow, and only the resulting credential is submitted to the extension.

Reply:
I'm not 100% sure that this is enough: [8787eae](https://github.com/KILTprotocol/spec-ext-credential-api/pull/57/commits/8787eae5fc46fc886a67533709fece0df5baac54).
Maybe we need to specify more? I thought about how the salts are generated. But maybe we leave all that open since this would be the most flexible for the issuer? Maybe at this point we also need to specify what are valid combinations of messages? E.g. it's not valid to request payment without submitting terms before that.

Reply:
I think it makes sense to add detail on who generates what parts of the credential. If there is an agreement, the issuer must base the credential they issue on the agreed upon data, including salts (if given?); If there is no agreement and the submit-credential is the only step, then the issuer creates all that data on their side, including salts.